src/main/scala/it/pagopa/interop/notifier/service/converters/PurposeEventsConverter.scala

```diff
@@ -3,57 +3,66 @@ package it.pagopa.interop.notifier.service.converters
+import it.pagopa.interop.notifier.model
+import it.pagopa.interop.notifier.model.{IndexRecord, MessageId, NotificationPayload, PurposePayload}
+import it.pagopa.interop.notifier.service.CatalogManagementService
 import it.pagopa.interop.notifier.service.converters.EventType._
-import it.pagopa.interop.notifier.service.{CatalogManagementService, DynamoService}
+import it.pagopa.interop.notifier.service.impl.DynamoIndexService
 import it.pagopa.interop.purposemanagement.model.persistence._
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object PurposeEventsConverter {
-  def getMessageId(catalogManagementService: CatalogManagementService, dynamoService: DynamoService)(implicit
+  def getMessageId(catalogManagementService: CatalogManagementService, dynamoIndexService: DynamoIndexService)(implicit
     ec: ExecutionContext,
     contexts: Seq[(String, String)]
   ): PartialFunction[ProjectableEvent, Future[MessageId]] = { case e: Event =>
-    getMessageIdFromEvent(catalogManagementService, dynamoService, e)
+    getMessageIdFromEvent(catalogManagementService, dynamoIndexService, e)
   }
 
   private[this] def getMessageIdFromEvent(
     catalogManagementService: CatalogManagementService,
-    dynamoService: DynamoService,
+    dynamoIndexService: DynamoIndexService,
     event: Event
   )(implicit ec: ExecutionContext, contexts: Seq[(String, String)]): Future[MessageId] = event match {
-    case PurposeCreated(purpose)             =>
-      catalogManagementService
-        .getEServiceProducerByEServiceId(purpose.eserviceId)
-        .map(organizationId => MessageId(purpose.id, organizationId))
-    case PurposeUpdated(purpose)             => getMessageIdFromDynamo(dynamoService)(purpose.id)
-    case PurposeVersionCreated(purposeId, _) => purposeId.toFutureUUID.flatMap(getMessageIdFromDynamo(dynamoService))
-    case PurposeVersionActivated(purpose)    => getMessageIdFromDynamo(dynamoService)(purpose.id)
-    case PurposeVersionSuspended(purpose)    => getMessageIdFromDynamo(dynamoService)(purpose.id)
-    case PurposeVersionWaitedForApproval(purpose) => getMessageIdFromDynamo(dynamoService)(purpose.id)
-    case PurposeVersionArchived(purpose)          => getMessageIdFromDynamo(dynamoService)(purpose.id)
-    case PurposeVersionUpdated(purposeId, _) => purposeId.toFutureUUID.flatMap(getMessageIdFromDynamo(dynamoService))
-    case PurposeVersionDeleted(purposeId, _) => purposeId.toFutureUUID.flatMap(getMessageIdFromDynamo(dynamoService))
-    case PurposeDeleted(purposeId)           => purposeId.toFutureUUID.flatMap(getMessageIdFromDynamo(dynamoService))
+    case PurposeCreated(purpose)                  =>
+      for {
+        messageId <- catalogManagementService
+          .getEServiceProducerByEServiceId(purpose.eserviceId)
+          .map(organizationId => model.MessageId(purpose.id, organizationId))
+        _         <- dynamoIndexService.put(IndexRecord(messageId))
```

Comment:
is this put idempotent? (thinking about retry in case of errors, or reprocesses)

Reply:
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html
In particular:

Reply:
https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_PutItem.html
In particular:
```
Creates a new item, or replaces an old item with a new item. 
If an item that has the same primary key as the new item already exists in the specified table, 
the new item completely replaces the existing item

Reply:
🔝